packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -990,10 +991,8 @@ export async function getBase(
 	const webhookWaitingBaseUrl = urlBaseWebhook + config.getEnv('endpoints.webhookWaiting');
 	const webhookTestBaseUrl = urlBaseWebhook + config.getEnv('endpoints.webhookTest');
 
-	const [encryptionKey, variables] = await Promise.all([
-		UserSettings.getEncryptionKey(),
-		WorkflowHelpers.getVariables(),
-	]);
+	const encryptionKey = await UserSettings.getEncryptionKey();
+	const variables = isVariablesEnabled() ? await WorkflowHelpers.getVariables() : {};
```

Comment:
@Joffcom pointed out 
> this seems like it will also break variables if the license expires, which will break workflows and that wasn’t something we wanted to do originally.

@krynble @valya opinions?

It just seems like a vast majority of users don't benefit from this query, but it still adds unnecessary delays in executions.
In queue-mode, even when this data is cached, it's still cached on redis, which adds delays as the traffic scales up.

Reply:
Maybe something like `isVariablesEnabled() || hasCachedVariables()`?

Reply:
Indeed as @Joffcom pointed this could break when license expires, but if this is a temporary issue (such as license server off) this is something we're handling by being very conservative when it comes to renewing licenses, giving it a decent grace period. And if the licenses do expire because in fact it's no longer valid, then it should in fact stop working, so I think this change is ok.

Reply:
Variables were indented to keep working after a license expired as long as the instance wasn't restarted. I can't find where the decision was made but it was intentional that they'd keep working after the license expired.